### PR TITLE
AISDK-48: Adding examples to the python SDK

### PR DESCRIPTION
**Pull request by Ymardini**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,3 +112,13 @@ Both the json and object forms contain all the formation outlined in the respons
...
+
+### Streaming audio
+
+Once you have a streaming client setup with a MediaConfig and access token, you can obtain a transcription generator of your audio.
+
+```python
+response_generator = streamingclient.start(AUDIO_GENERATOR)
+```
+
+`response_generator` is a generator object that yields the transcription results of the audio including partial and final transcriptions. The start method creates an thread sending audio pieces from the audio_generator to our Streaming]e endpoint.
```
Comment (kbridbur) [inline README.md]: typo `a thread`

Comment (kbridbur) [reply README.md]: typo `streaming`

File: README.md [hunk 2]
```diff
@@ -112,3 +112,13 @@ Both the json and object forms contain all the formation outlined in the respons
 of the [Get Transcript](https://www.rev.ai/docs#operation/GetTranscriptById) endpoint
 when using the json response schema. While the text output is a string containing
 just the text of your transcript
+
+### Streaming audio
+
+Once you have a streaming client setup with a MediaConfig and access token, you can obtain a transcription generator of your audio.
```
Comment (kbridbur) [inline README.md]: I would show how to do this and also include something about each of the functions `on_error`, `on_connected` and `on_close`

File: examples/async_example.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        break
+    break
+
+#Getting a list of current jobs connected with your account
+#The optional parameters limits limits the length of the list. Starting_after
+#Cuts off the beginning x jobs off the list returned
+list_of_jobs = client.get_list_of_jobs(limits=None, starting_after=None)
+
+#obtain transcript text as a string for the job.id.
```
Comment (kbridbur) [inline examples/async_example.py]: I would just say `for the job`. `job.id` sounds a bit weird here to me

File: examples/async_example.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+transcript_obj = client.get_transcript_object(job.id)
+
+#obtain captions for the job.id.
+captions = client.get_captions(job.id)
+
+#Use the objects however you please
+
+#Once you are done with the job, you can delete it.
+#NOTE : This will PERMANENTLY DELETE all data related to a job. Exercise only
+#if sure you want to delete the job.
```
Comment (kbridbur) [inline examples/async_example.py]: typo `if you're sure`

File: examples/generator_streaming_example.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+#String of your access token
+access_token = "your_access_token"
+
+#Media configuration of audio file. This includes the content type, layout, rate, format, and # of channels
+config = MediaConfig("audio/x-raw", "interleaved", 16000, "S16LE", 1)
+
+#Create client with your access token and media configuration
+streamclient = RevAiStreamingClient(access_token, config)
+
+#Open file and read data into array. Practically, stream data would be divded into chunks
```
Comment (kbridbur) [inline examples/generator_streaming_example.py]: typo `divided`

File: examples/async_example.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+import time
+from rev_ai import apiclient, streamingclient
+from rev_ai.models import MediaConfig
+
+#String containing your access token
```
Comment (kbridbur) [inline examples/async_example.py]: For all of these comments, please add a space between the `#` and the rest of the comment. It makes it much more readable

File: examples/generator_streaming_example.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+response_generator = streamclient.start(MEDIA_GENERATOR)
+
+#Iterates through the responses from the server when obtained
+for response in response_generator:
+    print(response)
+
+#Closes the streaming connection
+streamclient.end()
+
+########## Generator Example ##################################################
```
Comment (kbridbur) [inline examples/generator_streaming_example.py]: I'm not sure this example is needed, it seems redundant given the previous example

File: examples/generator_streaming_example.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+#Starts the streaming connection and creates a thread to send bytes from data_array
+#Response_generator is a generator yielding responses from the server
+response_generator = streamclient.start(MEDIA_GENERATOR)
+
+#Iterates through the responses from the server when obtained
+for response in response_generator:
+    print(response)
+
+#Closes the streaming connection
+streamclient.end()
```
Comment (kbridbur) [inline examples/generator_streaming_example.py]: I would note that the connection will normally be closed for the user and that this method is only to cut off the connection early

File: examples/microphone_streaming_example.py [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+            if len(message) < prev_message_length:
+                char_diff = prev_message_length - len(message)
+            else:
+                prev_message_length = len(message)
+
+            print(message + " " * char_diff)
+            prev_message_length = 0
+            char_diff = 0
+
+#Sampling rate of your microphone and desirable chunk size
```
Comment (kbridbur) [inline examples/microphone_streaming_example.py]: typo `desired`

File: examples/microphone_streaming_example.py [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                    data.append(chunk)
+                except queue.Empty:
+                    break
+
+            yield b''.join(data)
+
+
+def get_results(data_gen):
+    """Function to display interim responses from data_gen. Prints partial transcripts and
+        edits them as new responses change. Finalizes and makes a new line when recieving a
```
Comment (JohnDonavon) [inline examples/microphone_streaming_example.py]: --minor typo: recieving -> receiving

File: examples/generator_streaming_example.py [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+# Starts the streaming connection and creates a thread to send bytes from the MEDIA_GENERATOR
+# Response_generator is a generator yielding responses from the server
+response_generator = streamclient.start(MEDIA_GENERATOR)
+
+# Iterates through the responses from the server when obtained
+for response in response_generator:
+    print(response)
+
+# Ends the connection early. Not needed as the server will close the connetion
```
Comment (kbridbur) [inline examples/generator_streaming_example.py]: typo: `connection`

**Pull request merged**